integration-tests/app_id_test.go

```diff
@@ -40,23 +41,66 @@ func (suite *AppCredentialTestSuite) SetupSuite() {
 			{Month, 10000},
 		}},
 	}
-	serviceErr := suite.backend.Push("service", []interface{}{suite.ServiceID, suite.ServiceToken})
-	require.Nilf(suite.T(), serviceErr, "Error: %v", serviceErr)
-
-	appErr := suite.backend.Push("app", []interface{}{suite.ServiceID, suite.AppID, suite.PlanID})
-	require.Nilf(suite.T(), appErr, "Error: %v", appErr)
-
-	userErr := suite.backend.Push("user_key", []interface{}{suite.ServiceID, suite.AppID, suite.UserKey})
-	require.Nilf(suite.T(), userErr, "Error: %v", userErr)
-
-	appKeyErr := suite.backend.Push("app_key", []interface{}{suite.ServiceID, suite.AppID, suite.AppKey})
-	require.Nilf(suite.T(), appKeyErr, "Error: %v", appKeyErr)
-
-	metricsErr := suite.backend.Push("metrics", []interface{}{suite.ServiceID, &suite.metrics})
-	require.Nilf(suite.T(), metricsErr, "Error: %v", metricsErr)
 
-	usageErr := suite.backend.Push("usages", []interface{}{suite.ServiceID, suite.PlanID, &suite.metrics})
-	require.Nilf(suite.T(), usageErr, "Error: %v", usageErr)
+	require.Eventually(suite.T(), func() bool {
+		serviceErr := suite.backend.Push("service", []interface{}{suite.ServiceID, suite.ServiceToken})
+		if serviceErr != nil {
+			fmt.Printf("Error creating the service: %v", serviceErr)
+			return false
+		}
+		return true
+	}, 5*time.Second, 1*time.Second)
```

Comment:
I think eventually serves no purpose here since `Push` defers till the response is received and adding a 500ms wait before `Push` returns (I've added that already in my new PR) kind of guarantees that things are processed by the Apisonator.

Reply:
For things like timeouts and waits, it's better to use `require.Eventually` rather than adding `time.Sleep()` since the test suite already has this feature. It will perform retries periodically according to the way we define the related times. I'm thinking to remove the static `time.Sleep` in proxy start up method and do a check there.

Reply:
It wait for the function to return success but `cmd.Run` already returns success even though the proxy is still loading so eventually serves no purpose there as well since we are waiting for the proxy to get ready.

Reply:
yeah that's why we need to do a check using eventually to test whether the proxy is up or not.

Reply:
Can you explain to me how it does that?

Reply:
From the admin  check for the listener. 
```go
require.Eventually(t, func() bool {
    res, err := http.Get(fmt.Sprintf("http://localhost:%d/listeners", adminPort))
    if err != nil {
	    return false
    }
    defer res.Body.Close()
    return res.StatusCode == http.StatusOK
}, 5*time.Second, 100*time.Millisecond, "Envoy has not started")
```

Reply:
Better solution! Please add this into the PR itself.

Reply:
No need to have a event to check apisonator. Helper methods returns client errors, server errors and network errors and `Push` returns those errors directly to the caller. In case of such issues, eventually performs retries.